whitepaper/whitepaper.md

```diff
@@ -124,13 +134,235 @@ Blockchain technologies are growing in usage, but fragmentation is a big problem
 
 We propose a protocol and it's implementation to connect as many of them as possible in an attempt to solve the fragmentation problem by creating a heterogeneous system architecture <sup>[1](#7-references)</sup>.
 
-# 2. Example Use Cases
+# 2. Introduction to Blockchain Interoperability
+
+There are two inherent problems<sup>[a](#22-footnotes-introduction)</sup> that have to be solved when connecting different blockchains: 
+* How to provide a proof of the networkwide<sup>[b](#22-footnotes-introduction)</sup> ledger state of a connected blockchain from the outside?
+* How can other entities verify a given proof of the state of a connected blockchain
+from the outside? 
+
+The `Cactus` consortium operates for each connected blockchain a group of validator nodes, which as a group provides the proofs of the state of the connected ledger. The group of validator nodes runs a consensus algorithm to agree on the state of the underlying blockchain. Since a proof of the state of the blockchain is produced and signed by several validator nodes<sup>[c](#22-footnotes-introduction)</sup> with respect to the rules of the consensus algorithm, the state of the underlying blockchain is evaluated networkwide.
+
+The validator nodes are ledger-specific plug-ins, hence a smart contract on the connected blockchain can enable the ledger-specific functionalities necessary for a validator node to observe the ledger state to finalize a proof. The validator nodes are easier discovered from the outside than the blockchain nodes. Hence, the benefit of operating the `Cactus` network to enable blockchain interoperability relies on the fact that for any cross-blockchain interaction the same type of validator node signatures can be used. That means, the cross-blockchain interaction can be done canonically with the validator node signatures in  `Cactus` rather than having to deal with many different ledger-specific types of blockchain node signatures. 
```

Comment:
Rather than having to discover for each blockchain their corresponding nodes, it is easier to discover all cactus validator nodes and then canonically achieve the blockchain interoperability through the cactus validator nodes proof producing capabilities. Therefore only the functionalities for discovering cactus validator nodes are needed in contrast to implementing for each blockchain the functionalities to discover their nodes and enabling these nodes to produce verifiable proofs about their state. I don’t know exactly if Cactus supports node discovery (e.g. through gossip protocol) or has a registry for all the validator nodes currently or if this is planned in the future.

The ledger-specific plug-ins have to provide the capabilities to produce proofs about the state of the connected blockchain. Smart contracts “can enable” this functionality but any other method is valid as well as you mentioned. I used “can enable” for that purpose because it is not mandatory to use smart contracts for the ledger-specific plug-ins.

Reply:
Thanks for clarifying, agreed! We need to make sure the validators can produce such proofs